src/coffee/stockimport.coffee

```diff
@@ -349,15 +350,32 @@ class StockImport
     posts = _.map inventoryEntries, (entry) =>
       existingEntry = @_match(entry, existingEntries)
       if existingEntry?
-        synced = @sync.buildActions(entry, existingEntry)
-        if synced.shouldUpdate()
-          @client.inventoryEntries.byId(synced.getUpdateId()).update(synced.getUpdatePayload())
-        else
-          Promise.resolve statusCode: 304
+        @_updateInventory(entry, existingEntry)
       else
         @client.inventoryEntries.create(entry)
 
     debug 'About to send %s requests', _.size(posts)
-    Promise.all(posts)
+    Promise.map(posts, (p) -> p)
+
+  _updateInventory: (entry, existingEntry, tryCount = 1) =>
+    synced = @sync.buildActions(entry, existingEntry)
+    if synced.shouldUpdate()
+      @client.inventoryEntries.byId(synced.getUpdateId()).update(synced.getUpdatePayload())
+      .catch (err) =>
+        if (err.statusCode == 409)
+          debug "Got 409 error for entry #{JSON.stringify(entry)}, repeat the request "
+          + "for #{tryCount} times."
+          if (tryCount <= @max409Retries)
+            @client.inventoryEntries
+            .byId(synced.getUpdateId())
+            .fetch()
+            .then (result) =>
+              @_updateInventory(entry, result.body, tryCount + 1)
+          else
+            Promise.reject "Retry limit #{max409Retries} reached for stock #{JSON.stringify(entry)}"
```

Comment:
Instead of rejecting a string, I think it's better we reject an error object. Applies to other places too.

Reply:
https://github.com/sphereio/sphere-stock-import/pull/67/commits/27edbc3d47cc270c717f70485effc92329baee05